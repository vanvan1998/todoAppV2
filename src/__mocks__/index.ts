const todo = {
  email: 'nguyensivan1998+102@gmail.com',
  id: '82zB2At6gnxs0SbDJ8SW',
  notification: true,
  createdAt: '2024-05-10T01:14:38',
  title: '333333',
  detail: '3333333',
  startTime: '01:00:21',
  startDate: '05/10/2024',
  category: 1,
  completed: false,
  userId: 'j3WklyoTrBg4E1GKNz0Gn0az34V2',
  checkList: []
};
export { todo };
