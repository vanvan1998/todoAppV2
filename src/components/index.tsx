export { AddTodoModal } from './AddTodoModal';
export { Search } from './Search';
export { TodoItem } from './TodoItem';
export { Container } from './Container';
export { Header } from './Header';
export { AddTodo } from './addTodo';
export { SortBy } from './SortBy';
export { Input } from './input';
export { Button } from './button';
