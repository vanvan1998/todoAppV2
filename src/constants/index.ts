export const SORT_BY = {
  NEWEST: 'NEWEST',
  OLDEST: 'OLDEST',
  NONE: 'NONE'
};

export const MODE = {
  UPDATE: 'UPDATE',
  ADD: 'ADD'
};

export const DATE_FORMAT = 'MM/DD/YYYY';
export const TIME_FORMAT = 'HH:mm:ss';
