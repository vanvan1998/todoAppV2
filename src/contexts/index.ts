export * from './AppProvider';
export * from './GlobalLoadingProvider';
export * from './AuthProvider';
