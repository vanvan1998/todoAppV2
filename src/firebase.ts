import { initializeApp } from 'firebase/app';
import { getAuth, signInWithPopup } from 'firebase/auth';
import { getStorage } from 'firebase/storage';
import { getFirestore } from 'firebase/firestore';
import { GoogleAuthProvider } from 'firebase/auth';

export const app = initializeApp({
  apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
  authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
  databaseURL: process.env.NEXT_PUBLIC_FIREBASE_DATABASE_URL,
  projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
  storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID,
  appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID
});

const provider = new GoogleAuthProvider();
provider.setCustomParameters({ prompt: 'select_account' });

export const auth = getAuth(app);
export const storage = getStorage(app);
export const db = getFirestore(app);

export const signInWithGoogle = () => signInWithPopup(auth, provider);