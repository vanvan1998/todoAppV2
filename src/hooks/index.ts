export * from './useFirebase';
export * from './useNotification';
export * from './useMediaQuery';
