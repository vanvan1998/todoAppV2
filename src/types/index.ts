
export interface TodoItemType {
  id: string;
  title: string;
  detail: string;
  completed: boolean;
  startDate?: string;
  startTime?: string;
  notification?: boolean;
  createdAt: string;
}