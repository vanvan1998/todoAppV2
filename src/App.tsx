import React from 'react';
import { AuthProvider } from './contexts/AuthContext';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import {
  Signup,
  LeftMenu,
  Login,
  PrivateRoute,
  ForgotPassword,
  UpdatePassword,
  UpdateProfile,
  DashBoard
} from './modules';

function App() {
  return (
    <div style={{ minHeight: '100vh', display: 'flex' }}>
      <Router>
        <AuthProvider>
          <Routes>
            <Route
              path='/update-password'
              element={
                <PrivateRoute>
                  <UpdatePassword />
                </PrivateRoute>
              }
            />
            <Route
              path='/update-profile'
              element={
                <PrivateRoute>
                  <UpdateProfile />
                </PrivateRoute>
              }
            />
            <Route path='/signup' element={<Signup />} />
            <Route path='/login' element={<Login />} />
            <Route path='/forgot-password' element={<ForgotPassword />} />
            <Route
              path='/left-menu'
              element={
                <PrivateRoute>
                  <LeftMenu />
                </PrivateRoute>
              }
            />
            <Route
              path='/'
              element={
                <PrivateRoute>
                  <DashBoard />
                </PrivateRoute>
              }
            />
          </Routes>
        </AuthProvider>
      </Router>
    </div>
  );
}

export default App;
