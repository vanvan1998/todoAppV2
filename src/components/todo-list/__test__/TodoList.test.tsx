import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TodoList } from '..';
import { todo } from 'src/__mocks__';
import { CategoryType } from 'src/types';

jest.mock('next/navigation', () => {
  return {
    useRouter: jest.fn().mockReturnValue({
      push: jest.fn()
    })
  };
});

describe('Test TodoList component', () => {
  test.only('should match its snapshot', async () => {
    const component = render(
      <TodoList
        todoListCategoryType={CategoryType.Important}
        todoList={[todo, { ...todo, completed: true }]}
        handleCompleteTodo={jest.fn()}
        handleDeleteTodo={jest.fn()}
        handleUpdateTodo={jest.fn()}
        onDragEnd={jest.fn()}
        onDragStart={jest.fn()}
        onDragOver={jest.fn()}
        handleAddTodoItem={jest.fn()}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
