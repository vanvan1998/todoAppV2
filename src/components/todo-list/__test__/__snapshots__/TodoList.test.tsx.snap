// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodoList component should match its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-blLrKi jPROYq"
      >
        <div
          class="sc-ktEKgv ckszcg"
          draggable="true"
          id="82zB2At6gnxs0SbDJ8SW"
        >
          <div
            class="sc-gueXAH gRHGwY"
            style="font-weight: 500; padding-bottom: 6px;"
          >
            333333
          </div>
          <div
            class="sc-fGFvMH kCKAqd"
          >
            <div
              class="sc-gueXAH gRHGwY"
              style="font-size: 12px; color: red;"
            >
              Expired time: 
            </div>
            <div
              class="sc-gueXAH gRHGwY"
              style="font-weight: bold; font-size: 12px; color: red;"
            >
              05/10/2024
               
              01:00:21
            </div>
          </div>
          <div
            class="sc-hLseSp fDeqZQ"
            style="height: 70px; width: 100%;"
          >
            3333333
          </div>
          <div
            class="sc-hAtFlw WdYvR"
          >
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent; color: rgb(100, 89, 227);"
              type="button"
            >
              Task details
            </button>
            <div
              class="sc-fmSBHF gzwAvP"
            >
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent;"
                type="button"
              >
                <svg
                  color="#6459e3"
                  fill="currentColor"
                  height="24"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 16 16"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14c-3.314 0-6-2.686-6-6s2.686-6 6-6c3.314 0 6 2.686 6 6s-2.686 6-6 6z"
                  />
                </svg>
              </button>
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
                type="button"
              >
                <svg
                  color="#6459e3"
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                  />
                </svg>
              </button>
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
                type="button"
              >
                <svg
                  color="#6459e3"
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM13.4142 13.9997L15.182 15.7675L13.7678 17.1817L12 15.4139L10.2322 17.1817L8.81802 15.7675L10.5858 13.9997L8.81802 12.232L10.2322 10.8178L12 12.5855L13.7678 10.8178L15.182 12.232L13.4142 13.9997ZM9 4V6H15V4H9Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-ktEKgv ckszcg"
          draggable="true"
          id="82zB2At6gnxs0SbDJ8SW"
        >
          <div
            class="sc-gueXAH gRHGwY"
            style="font-weight: 500; padding-bottom: 6px;"
          >
            333333
          </div>
          <div
            class="sc-fGFvMH kCKAqd"
          >
            <div
              class="sc-gueXAH gRHGwY"
              style="font-size: 12px; color: red;"
            >
              Expired time: 
            </div>
            <div
              class="sc-gueXAH gRHGwY"
              style="font-weight: bold; font-size: 12px; color: red;"
            >
              05/10/2024
               
              01:00:21
            </div>
          </div>
          <div
            class="sc-hLseSp fDeqZQ"
            style="height: 70px; width: 100%;"
          >
            3333333
          </div>
          <div
            class="sc-hAtFlw WdYvR"
          >
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent; color: rgb(100, 89, 227);"
              type="button"
            >
              Task details
            </button>
            <div
              class="sc-fmSBHF gzwAvP"
            >
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent;"
                type="button"
              >
                <svg
                  color="#6459e3"
                  fill="currentColor"
                  height="24"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
                  />
                </svg>
              </button>
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
                type="button"
              >
                <svg
                  color="#6459e3"
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                  />
                </svg>
              </button>
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
                type="button"
              >
                <svg
                  color="#6459e3"
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM13.4142 13.9997L15.182 15.7675L13.7678 17.1817L12 15.4139L10.2322 17.1817L8.81802 15.7675L10.5858 13.9997L8.81802 12.232L10.2322 10.8178L12 12.5855L13.7678 10.8178L15.182 12.232L13.4142 13.9997ZM9 4V6H15V4H9Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div>
          <button
            buttontype="secondary"
            class="sc-gLDymS jcdnJc"
            style="height: 170px; padding: 0px 16px; width: 100%; display: flex; align-items: center; justify-content: center; border-style: dashed; background-color: rgb(242, 246, 250);"
            type="button"
          >
            <svg
              fill="currentColor"
              height="32"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-blLrKi jPROYq"
    >
      <div
        class="sc-ktEKgv ckszcg"
        draggable="true"
        id="82zB2At6gnxs0SbDJ8SW"
      >
        <div
          class="sc-gueXAH gRHGwY"
          style="font-weight: 500; padding-bottom: 6px;"
        >
          333333
        </div>
        <div
          class="sc-fGFvMH kCKAqd"
        >
          <div
            class="sc-gueXAH gRHGwY"
            style="font-size: 12px; color: red;"
          >
            Expired time: 
          </div>
          <div
            class="sc-gueXAH gRHGwY"
            style="font-weight: bold; font-size: 12px; color: red;"
          >
            05/10/2024
             
            01:00:21
          </div>
        </div>
        <div
          class="sc-hLseSp fDeqZQ"
          style="height: 70px; width: 100%;"
        >
          3333333
        </div>
        <div
          class="sc-hAtFlw WdYvR"
        >
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent; color: rgb(100, 89, 227);"
            type="button"
          >
            Task details
          </button>
          <div
            class="sc-fmSBHF gzwAvP"
          >
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                version="1.1"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14c-3.314 0-6-2.686-6-6s2.686-6 6-6c3.314 0 6 2.686 6 6s-2.686 6-6 6z"
                />
              </svg>
            </button>
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                />
              </svg>
            </button>
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM13.4142 13.9997L15.182 15.7675L13.7678 17.1817L12 15.4139L10.2322 17.1817L8.81802 15.7675L10.5858 13.9997L8.81802 12.232L10.2322 10.8178L12 12.5855L13.7678 10.8178L15.182 12.232L13.4142 13.9997ZM9 4V6H15V4H9Z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-ktEKgv ckszcg"
        draggable="true"
        id="82zB2At6gnxs0SbDJ8SW"
      >
        <div
          class="sc-gueXAH gRHGwY"
          style="font-weight: 500; padding-bottom: 6px;"
        >
          333333
        </div>
        <div
          class="sc-fGFvMH kCKAqd"
        >
          <div
            class="sc-gueXAH gRHGwY"
            style="font-size: 12px; color: red;"
          >
            Expired time: 
          </div>
          <div
            class="sc-gueXAH gRHGwY"
            style="font-weight: bold; font-size: 12px; color: red;"
          >
            05/10/2024
             
            01:00:21
          </div>
        </div>
        <div
          class="sc-hLseSp fDeqZQ"
          style="height: 70px; width: 100%;"
        >
          3333333
        </div>
        <div
          class="sc-hAtFlw WdYvR"
        >
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent; color: rgb(100, 89, 227);"
            type="button"
          >
            Task details
          </button>
          <div
            class="sc-fmSBHF gzwAvP"
          >
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
                />
              </svg>
            </button>
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                />
              </svg>
            </button>
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM13.4142 13.9997L15.182 15.7675L13.7678 17.1817L12 15.4139L10.2322 17.1817L8.81802 15.7675L10.5858 13.9997L8.81802 12.232L10.2322 10.8178L12 12.5855L13.7678 10.8178L15.182 12.232L13.4142 13.9997ZM9 4V6H15V4H9Z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div>
        <button
          buttontype="secondary"
          class="sc-gLDymS jcdnJc"
          style="height: 170px; padding: 0px 16px; width: 100%; display: flex; align-items: center; justify-content: center; border-style: dashed; background-color: rgb(242, 246, 250);"
          type="button"
        >
          <svg
            fill="currentColor"
            height="32"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
